From 1f2095e09188d31c36957889cef9683ca534db4d Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Thu, 18 Mar 2010 17:38:14 -0500
Subject: [PATCH] ENGR00121762-1 mx5: enable run-time physoffset

This enables a single kernel image on MX5x parts using run-time
phys_offset.

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 arch/arm/configs/imx5_defconfig         |   19 ++++++++++++++++---
 arch/arm/mach-mx5/Kconfig               |    6 ++++++
 arch/arm/plat-mxc/include/mach/memory.h |    2 ++
 3 files changed, 24 insertions(+), 3 deletions(-)

diff --git a/arch/arm/configs/imx5_defconfig b/arch/arm/configs/imx5_defconfig
index 16eb284..d5dbd32 100644
--- a/arch/arm/configs/imx5_defconfig
+++ b/arch/arm/configs/imx5_defconfig
@@ -195,10 +195,12 @@ CONFIG_MXC_SDMA_API=y
 CONFIG_SDMA_IRAM=y
 CONFIG_FORCE_MAX_ZONEORDER=13
 CONFIG_ARCH_MXC_HAS_NFC_V3=y
+CONFIG_ARCH_MX51=y
 CONFIG_ARCH_MX53=y
 CONFIG_MX5_OPTIONS=y
-# CONFIG_MACH_MX51_3DS is not set
-# CONFIG_MACH_MX51_BABBAGE is not set
+CONFIG_MX5_MULTI_ARCH=y
+CONFIG_MACH_MX51_3DS=y
+CONFIG_MACH_MX51_BABBAGE=y
 CONFIG_MACH_MX53_EVK=y
 
 #
@@ -284,6 +286,7 @@ CONFIG_DEFAULT_MMAP_MIN_ADDR=4096
 # CONFIG_LEDS is not set
 CONFIG_ALIGNMENT_TRAP=y
 # CONFIG_UACCESS_WITH_MEMCPY is not set
+CONFIG_RUNTIME_PHYS_OFFSET=y
 
 #
 # Boot options
@@ -536,7 +539,7 @@ CONFIG_MTD_MXC_DATAFLASH=y
 # CONFIG_MTD_DOC2000 is not set
 # CONFIG_MTD_DOC2001 is not set
 # CONFIG_MTD_DOC2001PLUS is not set
-CONFIG_MTD_NAND=y
+# CONFIG_MTD_NAND is not set
 # CONFIG_MTD_NAND_VERIFY_WRITE is not set
 # CONFIG_MTD_NAND_ECC_SMC is not set
 # CONFIG_MTD_NAND_MUSEUM_IDS is not set
@@ -799,6 +802,7 @@ CONFIG_HW_CONSOLE=y
 # CONFIG_VT_HW_CONSOLE_BINDING is not set
 CONFIG_DEVKMEM=y
 # CONFIG_SERIAL_NONSTANDARD is not set
+CONFIG_FM_SI4702=m
 CONFIG_MXC_IIM=y
 CONFIG_IMX_SIM=m
 
@@ -883,6 +887,7 @@ CONFIG_SPI_MXC=y
 # CONFIG_SPI_MXC_TEST_LOOPBACK is not set
 CONFIG_SPI_MXC_SELECT1=y
 # CONFIG_SPI_MXC_SELECT2 is not set
+# CONFIG_SPI_MXC_SELECT3 is not set
 
 #
 # SPI Protocol Masters
@@ -1007,6 +1012,7 @@ CONFIG_HWMON=y
 # CONFIG_SENSORS_W83627EHF is not set
 # CONFIG_SENSORS_LIS3_SPI is not set
 # CONFIG_HWMON_DEBUG_CHIP is not set
+CONFIG_SENSORS_ISL29003=y
 # CONFIG_THERMAL is not set
 # CONFIG_THERMAL_HWMON is not set
 CONFIG_WATCHDOG=y
@@ -1443,6 +1449,10 @@ CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_EHCI_ARC=y
 CONFIG_USB_EHCI_ARC_OTG=y
 # CONFIG_USB_STATIC_IRAM is not set
+# CONFIG_USB_EHCI_FSL_MC13783 is not set
+# CONFIG_USB_EHCI_FSL_1301 is not set
+# CONFIG_USB_EHCI_FSL_1504 is not set
+CONFIG_USB_EHCI_FSL_UTMI=y
 CONFIG_USB_EHCI_ROOT_HUB_TT=y
 # CONFIG_USB_EHCI_TT_NEWSCHED is not set
 # CONFIG_USB_OXU210HP_HCD is not set
@@ -1536,6 +1546,7 @@ CONFIG_MMC_BLOCK=y
 CONFIG_MMC_BLOCK_BOUNCE=y
 # CONFIG_SDIO_UART is not set
 # CONFIG_MMC_TEST is not set
+CONFIG_SDIO_UNIFI_FS=m
 
 #
 # MMC/SD/SDIO Host Controller Drivers
@@ -1724,10 +1735,12 @@ CONFIG_MXC_VPU_IRAM=y
 #
 # MXC Bluetooth support
 #
+CONFIG_MXC_BLUETOOTH=m
 
 #
 # Broadcom GPS ioctrl support
 #
+CONFIG_GPS_IOCTRL=m
 
 #
 # MXC Media Local Bus Driver
diff --git a/arch/arm/mach-mx5/Kconfig b/arch/arm/mach-mx5/Kconfig
index e7bc97f..5c07a3a 100644
--- a/arch/arm/mach-mx5/Kconfig
+++ b/arch/arm/mach-mx5/Kconfig
@@ -17,6 +17,12 @@ config MX5_OPTIONS
 	select USB_ARCH_HAS_EHCI
 	select MXC_TZIC
 
+config MX5_MULTI_ARCH
+	bool
+	default y
+	select RUNTIME_PHYS_OFFSET
+	depends on ARCH_MX51 && ARCH_MX53
+
 config MACH_MX51_3DS
 	bool "Support MX51 3-Stack platform"
 	select ARCH_MX51
diff --git a/arch/arm/plat-mxc/include/mach/memory.h b/arch/arm/plat-mxc/include/mach/memory.h
index a70d3ef..c9609c4 100644
--- a/arch/arm/plat-mxc/include/mach/memory.h
+++ b/arch/arm/plat-mxc/include/mach/memory.h
@@ -14,6 +14,7 @@
 #include <asm/page.h>
 #include <asm/sizes.h>
 
+#ifndef CONFIG_RUNTIME_PHYS_OFFSET
 #if defined CONFIG_ARCH_MX1
 #define PHYS_OFFSET		UL(0x08000000)
 #elif defined CONFIG_ARCH_MX2
@@ -42,6 +43,7 @@
 #ifndef PHYS_OFFSET
 #define PHYS_OFFSET	        UL(0x80000000)
 #endif
+#endif
 
 /* Size of contiguous memory for DMA and other h/w blocks */
 #if defined(CONFIG_MX1_VIDEO)
-- 
1.5.4.4

